config/clients/python/template/configuration.handlebars

```diff
@@ -0,0 +1,590 @@
+        :param servers: an array of host settings or None
+        :return: URL based on host settings
+        """
+        if index is None:
+            return self._base_path
+
+        variables = {} if variables is None else variables
+        servers = self.get_host_settings() if servers is None else servers
+
+        try:
+            server = servers[index]
+        except IndexError:
+            raise ValueError(
+                "Invalid index {0} when selecting the host settings. "
+                "Must be less than {1}".format(index, len(servers)))
+
+        url = server['url']
+
+        # go through variables and replace placeholders
+        for variable_name, variable in server.get('variables', {}).items():
+            used_value = variables.get(
+                variable_name, variable['default_value'])
+
+            if 'enum_values' in variable \
+                    and used_value not in variable['enum_values']:
+                raise ValueError(
+                    "The variable `{0}` in the host URL has invalid value "
+                    "{1}. Must be {2}.".format(
+                        variable_name, variables[variable_name],
+                        variable['enum_values']))
+
+            url = url.replace("{" + variable_name + "}", used_value)
+
+        return url
+
+    def is_valid(self):
+        """
+        Verify the configuration is valid.
+        Note that we are only doing basic validation to ensure input is sane.
+        """
+        if self.api_host is None or self.api_host == '':
+            raise ApiTypeError('api_host is required but not configured.')
+        if self.api_scheme is None or self.api_scheme == '':
+            raise ApiTypeError('api_scheme is required but not configured.')
+        combined_url = self.api_scheme + '://' + self.api_host
+        parsed_url = None
+        try:
+            parsed_url = urlparse(combined_url)
+        except:
+            raise ApiValueError('Either api_scheme `{}` or api_host `{}` is invalid'.format(self.api_scheme, self.api_host))
+        if (parsed_url.scheme != 'http' and parsed_url.scheme != 'https'):
+            raise ApiValueError('api_scheme `{}` must be either `http` or `https`'.format(self.api_scheme))
+        if (parsed_url.netloc == ''):
+            raise ApiValueError('api_host `{}` is invalid'.format(self.api_host))
+        if (parsed_url.path != ''):
+            raise ApiValueError('api_host `{}` is not expected to have path specified'.format(self.api_scheme))
+        if (parsed_url.query != ''):
+            raise ApiValueError('api_host `{}` is not expected to have query specified'.format(self.api_scheme))
+        if self._credentials is not None:
+            self._credentials.validate_credentials_config()
+
+    @property
+    def api_scheme(self):
+        """Return connection is https or http."""
+        return self._scheme
+
+    @api_scheme.setter
```

Comment:
Can we add support for passing in custom headers? And maybe useragent can be defined here?

Reply:
If the user has passed in a custom user agent header, we shouldn't override

Reply:
We already did - see https://github.com/openfga/sdk-generator/blob/72a04e6e1d30c5439b13ee5a67ccf2fc19584ff1/config/clients/python/template/api_client.handlebars#L131 and https://github.com/openfga/sdk-generator/blob/72a04e6e1d30c5439b13ee5a67ccf2fc19584ff1/config/clients/python/template/api_client.handlebars#L128

Reply:
I will add unit test to confirm we did that.